game-frontend/src/components/canvas-objects/GameObjects.tsx

```diff
@@ -10,17 +16,37 @@ type Props = {
 
 const GameObjects: FC<Props> = ({ objectIds, objectMap, scaleFactor }) => (
   <>
-    {objectIds.map(objectId => {
-      const { position } = objectMap[objectId];
-      return (
-        <Unit
-          key={objectId}
-          position={{
-            x: position.x * scaleFactor,
-            y: position.y * scaleFactor,
-          }}
-        />
-      );
+    {objectIds.map((objectId): JSX.Element | null => {
+      const gameObject = objectMap[objectId];
+      switch (gameObject.type) {
+        case GAME_OBJECT_TYPES.UNIT: {
+          const unit = gameObject as UnitT;
+          return (
+            <Unit
+              key={objectId}
+              health={unit.health}
+              position={{
+                x: unit.position.x * scaleFactor,
+                y: unit.position.y * scaleFactor,
+              }}
+            />
+          );
+        }
+        case GAME_OBJECT_TYPES.FOOD: {
+          const food = gameObject as UnitT;
+          return (
+            <Food
+              key={objectId}
+              position={{
+                x: food.position.x * scaleFactor,
+                y: food.position.y * scaleFactor,
+              }}
+            />
+          );
+        }
```

Comment:
I'm wondering if there's a way to generalize that. This list will grow large quite fast.

Reply:
Then it will be split up in separate files :)

Reply:
Okay, let's leave it for now. We can still refactor it later :)